<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Shader_Dlg</class>
 <widget class="QWidget" name="Shader_Dlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1076</width>
    <height>632</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QTabWidget" name="tabWidget_shaderPages">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>21</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <property name="documentMode">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="tab_std">
        <attribute name="title">
         <string>Std</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="tab_shader">
        <attribute name="title">
         <string>Shader</string>
        </attribute>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QSplitter" name="splitter">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <widget class="QWidget" name="">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="CodeEditor" name="plainTextEdit_editor">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="pushButton_undo">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="focusPolicy">
              <enum>Qt::NoFocus</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="D3DChips_Dlg.qrc">
               <normaloff>:/D3DChips_Dlg/Resources/Edit_UndoHS.png</normaloff>:/D3DChips_Dlg/Resources/Edit_UndoHS.png</iconset>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_redo">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="focusPolicy">
              <enum>Qt::NoFocus</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="D3DChips_Dlg.qrc">
               <normaloff>:/D3DChips_Dlg/Resources/Edit_RedoHS.png</normaloff>:/D3DChips_Dlg/Resources/Edit_RedoHS.png</iconset>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_line">
             <property name="minimumSize">
              <size>
               <width>60</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>Ln 1</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_character">
             <property name="minimumSize">
              <size>
               <width>60</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>Ch 1</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QPlainTextEdit" name="plainTextEdit_msg">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line_4">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <item>
      <widget class="QComboBox" name="comboBox_shaderType">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>288</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Compile</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox_shaderModel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox_optimization">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <item>
        <property name="text">
         <string>Disable Optimizations</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Optimization Level 0</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Optimization Level 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Optimization Level 2</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Optimization Level 3</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox_flowControl">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <item>
        <property name="text">
         <string>Default Flow Control</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Prefer Flow Control</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Avoid Flow Control</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkBox_debug">
       <property name="text">
        <string>Debug Compile</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkBox_forceHalf">
       <property name="text">
        <string>Force Half Precision</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>6</number>
       </property>
       <item>
        <widget class="QPushButton" name="pushButton_compile">
         <property name="text">
          <string>Compile</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_compileC">
         <property name="text">
          <string>Compile to C-array</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>codeeditor.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="D3DChips_Dlg.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_undo</sender>
   <signal>clicked()</signal>
   <receiver>plainTextEdit_editor</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>33</x>
     <y>361</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_redo</sender>
   <signal>clicked()</signal>
   <receiver>plainTextEdit_editor</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>361</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>plainTextEdit_editor</sender>
   <signal>cursorPositionChanged()</signal>
   <receiver>Shader_Dlg</receiver>
   <slot>onCursorPositionChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>423</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>345</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_shaderType</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>Shader_Dlg</receiver>
   <slot>onShaderTypeSelected()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>345</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>plainTextEdit_editor</sender>
   <signal>undoAvailable(bool)</signal>
   <receiver>pushButton_undo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>423</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>33</x>
     <y>361</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>plainTextEdit_editor</sender>
   <signal>redoAvailable(bool)</signal>
   <receiver>pushButton_redo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>423</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>67</x>
     <y>361</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_compile</sender>
   <signal>clicked()</signal>
   <receiver>Shader_Dlg</receiver>
   <slot>onCompile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>788</x>
     <y>613</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>345</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget_shaderPages</sender>
   <signal>currentChanged(int)</signal>
   <receiver>Shader_Dlg</receiver>
   <slot>shaderPageChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>345</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_compileC</sender>
   <signal>clicked()</signal>
   <receiver>Shader_Dlg</receiver>
   <slot>onCompileToC()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>791</x>
     <y>659</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>345</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onCursorPositionChanged()</slot>
  <slot>onShaderTypeSelected()</slot>
  <slot>onManualModeChanged()</slot>
  <slot>onEditorUpdated()</slot>
  <slot>onCompile()</slot>
  <slot>shaderPageChanged()</slot>
  <slot>onCompileToC()</slot>
 </slots>
</ui>
